util.c

```diff
@@ -153,3 +153,22 @@ map_physical(
 
         return (void *)(addr + page_offset);
 }
+
+void
+copy_physical(
+	uint64_t phys_addr,
+	size_t len,
+	volatile void * dest
+)
+{
+	const uint8_t * const buf = map_physical(phys_addr, len);
+	if (buf == NULL)
+	{
+		perror("mmap");
+		exit(EXIT_FAILURE);
+	}
+
+	memcpy(dest, buf, len);
+
+	munmap((uint8_t *)buf, len);
```

Comment:
This isn't guaranteed to unmap all of the mapped memory.  If phys_addr isn't page aligned or len is not a multiple of page length, map_physical() will fix them up and map a larger amount of memory.

Reply:
Good catch - I'll fix this up.